### ISSUE TITLE
An error is occured when lint @apply rules

### ISSUE THREAD

**Issue by morishitter** [stylelint/stylelint#624]
I tried to lint CSS code that uses [@apply rules](http://tabatkins.github.io/specs/css-apply-rule/) by [postcss-apply](https://github.com/pascalduez/postcss-apply), but the following error is occured.

```
Message:
    Cannot read property '0' of null
```

and here is my CSS code.

```css
:root {
  --truncate: {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
```

I wish your help. Thanks.

**Comment by jeddy3** [comment]
Hi @morishitter 

I was able to lint the following CSS without errors:

```css
:root {
  --toolbar-theme: {
    background-color: rebeccapurple;
    color: white;
    border: 1px solid green;
  };
  --toolbar-title-theme: {
    color: green;
  };
}

.toolbar {
  @apply --toolbar-theme;
}
.toolbar__title {
  @apply --toolbar-title-theme;
}
```

Can you paste in your config and let us know what version of stylelint you're using please, so that we can get to the bottom of this for you? :)

**Comment by morishitter** [comment]
@jeddy3 I'm using v3.2.2.

Here is my config file.

```js
var propertiesOrder = require('./properties-order');

module.exports = {
  "rules": {
    "color-hex-case": "lower",
    "color-hex-length": "short",
    "color-no-invalid-hex": true,
    "media-feature-name-no-vendor-prefix": true,
    "number-leading-zero": "always",
    "number-no-trailing-zeros": true,
    "number-zero-length-no-unit": true,
    "property-no-vendor-prefix": true,
    "rule-properties-order": propertiesOrder,
    "selector-no-vendor-prefix": true,
    "selector-pseudo-element-colon-notation": "double",
    "selector-root-no-composition": true
  },
}
```

properties-order.js:

```js
module.exports = [
  "box-sizing",
  "position",
  "top",
  "right",
  "bottom",
  "left",
  "z-index",
  "display",
  "visibility",
  "flex-direction",
  "flex-order",
  "flex-pack",
  "align-items",
  "justify-content",
  "float",
  "clear",
  "flex-align",
  "clip",
  "width",
  "height",
  "max-width",
  "max-height",
  "padding",
  "padding-top",
  "padding-right",
  "padding-bottom",
  "padding-left",
  "margin",
  "margin-top",
  "margin-right",
  "margin-bottom",
  "margin-left",
  "outline",
  "outline-width",
  "outline-style",
  "outline-color",
  "outline-offset",
  "border",
  "border-top",
  "border-right",
  "border-bottom",
  "border-left",
  "border-spacing",
  "border-collapse",
  "border-width",
  "border-style",
  "border-color",
  "border-radius",
  "overflow",
  "overflow-x",
  "overflow-y",
  "z-index",
]
```

Thanks :)

**Comment by jeddy3** [comment]
Thanks @morishitter 

I've narrowed the bug down to `selector-no-vendor-prefix`, in particular [this line](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-vendor-prefix/index.js#L25). 

It mistakenly thinks `--toolbar-theme: {}` is a pseudo selector.

We'll fix the bug, but in the meantime you can disable the `selector-no-vendor-prefix` to get linting again :)

**Comment by morishitter** [comment]
@jeddy3 Thank you for your support :D

**Comment by jeddy3** [comment]
@morishitter FYI, released as `3.2.3`

**Comment by morishitter** [comment]
@jeddy3 :+1:
